src/mpyl/run_plan.py

```diff
@@ -0,0 +1,27 @@
+"""This module contains the RunPlan class."""
+
+from dataclasses import dataclass
+
+from .project import Stage
+from .project_execution import ProjectExecution
+
+
+@dataclass(frozen=True)
+class RunPlan:
+    plan: dict[Stage, set[ProjectExecution]]
+
+    @staticmethod
+    def empty() -> "RunPlan":
```

Comment:
I have not seen this `"RunPlan"` syntax before, what does it do?

Reply:
Python is not very good at dealing with forward references (i.e. using types that haven't been defined yet), so in order to deal with this we need to quote the type hint to make it a literal and tell the interpreter to resolve it later. You can find this explained in more detail in https://peps.python.org/pep-0484/#forward-references and https://peps.python.org/pep-0484/#the-problem-of-forward-declarations.